packages/application-shell/src/components/application-page-title/application-page-title.spec.tsx

```diff
@@ -0,0 +1,85 @@
+  ${`/almond-40/settings/product-types/<id>`}                                   | ${`Settings - almond-40 - Merchant Center`}
+  ${`/almond-40/settings/product-types/<id>/attribute-definitions/detail/<id>`} | ${`Settings - almond-40 - Merchant Center`}
+  ${`/account/projects`}                                                        | ${`Account - Merchant Center`}
+  ${`/account/organizations`}                                                   | ${`Account - Merchant Center`}
+  ${`/account/organizations/<id>/teams`}                                        | ${`Account - Merchant Center`}
+  ${`/account/organizations/<id>/teams/<id>/permissions/<id>`}                  | ${`Account - Merchant Center`}
+  ${`/account/organizations/<id>/custom-applications`}                          | ${`Account - Merchant Center`}
+  ${`/account/organizations/<id>/custom-applications/owned/<id>`}               | ${`Account - Merchant Center`}
+  ${`/account/organizations/<id>/custom-applications/installations/<id>`}       | ${`Account - Merchant Center`}
+  ${`/login`}                                                                   | ${`Login - Merchant Center`}
+  ${`/login/new`}                                                               | ${`Login - Merchant Center`}
+  ${`/login/sso`}                                                               | ${`Login - Merchant Center`}
+  ${`/login/forgot`}                                                            | ${`Login - Merchant Center`}
+  ${`/login/choose`}                                                            | ${`Login - Merchant Center`}
+`(`when location is "$route"`, ({ route, title }) => {
+  it('should render default page title', async () => {
+    render(
+      <Router history={memoryHistory(route)}>
+        <ApplicationPageTitle />
+      </Router>
+    );
+    await waitFor(() => {
+      expect(document.title).toBe(title);
+    });
+  });
+});
+
+describe('when using the custom "content" to override default mapping', () => {
+  it('should render the custom value', async () => {
+    render(
+      <Router history={memoryHistory('/login/new')}>
+        <ApplicationPageTitle content={['Create a new account']} />
+      </Router>
+    );
+    await waitFor(() => {
+      expect(document.title).toBe(
+        'Create a new account - Login - Merchant Center'
+      );
+    });
+  });
+
+  describe('when the page title is too long', () => {
+    it('should render a truncated readable page title', async () => {
+      render(
+        <Router
+          history={memoryHistory(
+            '/almond-40/products/productid12345/variants/variantid12345'
+          )}
+        >
+          <ApplicationPageTitle
+            content={['Some product very long product name']}
+          />
+        </Router>
+      );
+      await waitFor(() => {
+        expect(document.title).toBe(
+          'Some product...product name - Products - almond-40 - Merchant Center'
+        );
+      });
+    });
+  });
```

Comment:
Nit: We can also use the `describe.each` here and have the test data in the "table" format.